support/cas-server-support-oidc-core-api/src/main/java/org/apereo/cas/oidc/web/OidcCallbackAuthorizeViewResolver.java

```diff
@@ -30,8 +34,16 @@ public ModelAndView resolve(final JEEContext ctx, final ProfileManager manager,
                 LOGGER.trace("Redirecting to URL [{}] without prompting for login", url);
                 return new ModelAndView(url);
             }
-            LOGGER.warn("Unable to detect an authenticated user profile for prompt-less login attempts");
-            return new ModelAndView(new RedirectView(OidcAuthorizationRequestSupport.getRedirectUrlWithError(ctx.getFullRequestURL(), OidcConstants.LOGIN_REQUIRED)));
+            val originalRedirectUrl = ctx.getRequestParameter(OAuth20Constants.REDIRECT_URI);
+
+            if (originalRedirectUrl.isEmpty()) {
+                val model = new HashMap<String, String>();
+                model.put(OAuth20Constants.ERROR, OidcConstants.LOGIN_REQUIRED);
+                return new ModelAndView(new MappingJackson2JsonView(), model);
+            } else {
+                LOGGER.warn("Unable to detect an authenticated user profile for prompt-less login attempts. Redirecting to URL[{}]", originalRedirectUrl.get());
+                return new ModelAndView(new RedirectView(OidcAuthorizationRequestSupport.getRedirectUrlWithError(originalRedirectUrl.get(), OidcConstants.LOGIN_REQUIRED)));
```

Comment:
Please break this apart into two lines; extract `OidcAuthorizationRequestSupport.getRedirectUrlWithError(originalRedirectUrl.get(), OidcConstants.LOGIN_REQUIRED)` into a separate variable.